source/Nevermore/AST/CustomExpression.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace Nevermore.AST
+{
+    public class CustomExpression : IExpression
```

Comment:
This class feels like a "do whatever you want" class. Do we need this? If we're going to force an expression structure for the SQL generation then we shouldn't have this class. Otherwise why not just use this for all SQL generation?

Reply:
It's a back door (that we currently have btw!) to allow you to do things that nevermore doesn't yet support. It is just a more specific backdoor than we currently have :) Ideally we wouldn't have this backdoor at all but that involves more work to support a bunch of extra features. I don't think we have the capacity to do this work right now... better to move partially in the right direction then not at all.

>Otherwise why not just use this for all SQL generation?

There's static typing benefits whereever we can use our non-custom expression classes.